### PR TITLE
Add CrossAttention operator

### PR DESCRIPTION
**Pull request by tianleiwu**
### Description

Move separated Q, K and V (without input projection) from Attention to a new operator CrossAttention.

The Attention operator is hard to maintain when we need support with and without input projection in one class. Add a new operator according to feedback.

Some change might need in the future, but not in this PR:
(1) bias could be optional (We will not proceed that route unless experiments show that fusing Add bias with MatMul instead of this op could improve performance).
(2) support packed KV. There are two ways to support it:  when key and value are same Tensor, they are packed; or we can make value as optional, and use packed mode when value is empty and the key has packed K/V.
(3) support cached key and value, and other (like relative position bias), or more attention mask format. They can be added easily without breaking backward compatible.
(4) ROCm/CPU implementation of this op.

### CODE REVIEW

**Review by yufenglee — APPROVED** [review]
:shipit:

File: onnxruntime/core/graph/contrib_ops/bert_defs.cc [hunk 1]
```diff
@@ -246,6 +258,50 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
...
+               "T")
+        .Input(1,
+               "key",
+               "Key with shape (batch_size, kv_sequence_length, hidden_size)",
+               "T")
+        .Input(2,
+               "value",
+               "Value with shape (batch_size, kv_sequence_length, v_hidden_size)",
+               "T")
+        .Input(3,
```
Comment (wangyems) [inline onnxruntime/core/graph/contrib_ops/bert_defs.cc:286]: optional?

File: onnxruntime/core/graph/contrib_ops/shape_inference_functions.cc [hunk 2]
```diff
@@ -113,16 +113,10 @@ void AttentionTypeAndShapeInference(ONNX_NAMESPACE::InferenceContext& ctx, int p
...
-  //    INput 1 has 2D shape (input_hidden_size, hidden_size + hidden_size + v_hidden_size)
-  //    Input 2 has 1D shape (hidden_size + hidden_size + v_hidden_size)
-  // When weights is not avaiable (supported by Attention but not in QAttention):
-  //    Input 0 (query) has 3D shape (batch_size, sequence_length, hidden_size)
-  //    Input 2 (bias) has 1D shape (hidden_size + hidden_size + v_hidden_size)
-  //    Input 4 (past) has shape (2, batch_size, num_heads, past_sequence_length, head_size)
-  //    Input 6 (value) has shape (batch_size, kv_sequence_length, v_hidden_size)
-  //
+  // Input 0 has 3D shape (batch_size, sequence_length, input_hidden_size)
+  // INput 1 has 2D shape (input_hidden_size, hidden_size + hidden_size + v_hidden_size)
```
Comment (wangyems) [inline onnxruntime/core/graph/contrib_ops/shape_inference_functions.cc:117]: Input

**Review by wangyems — APPROVED** [review]

File: onnxruntime/core/graph/contrib_ops/bert_defs.cc (hunk 1 above)
Comment (tianleiwu) [reply onnxruntime/core/graph/contrib_ops/bert_defs.cc:286]: Packed kv will be supported later. Will update the interface accordingly at that time.

File: onnxruntime/core/graph/contrib_ops/shape_inference_functions.cc (hunk 2 above)
Comment (tianleiwu) [reply onnxruntime/core/graph/contrib_ops/shape_inference_functions.cc:117]: will fix it in next PR.

**Pull request merged**